src/caikit_nlp_client/http_client.py

```diff
@@ -83,9 +90,20 @@ def _get_tls_configuration(self) -> dict[str, Union[str, tuple[str, str]]]:
                 self._client_cert_path,
                 self._client_key_path,
             )
+
         if self._ca_cert_path:
             req_kwargs["verify"] = self._ca_cert_path
 
+        if self._insecure is not None:
```

Comment:
This is what codecov is complaining about. We should never hit this because of the check in `__init__`. I'm open to removing this check.

Reply:
I am ok with the removal of the check.